### ISSUE TITLE
ActiveRecord Magic and ?? operator

### ISSUE THREAD

**Issue by paulzi** [yiisoft/yii2#16653]
### What steps will reproduce the problem?

```php
class Item extends yii\db\ActiveRecord {
    public function getList()
    {
        return [1, 2, 3 / 0]; // division by zero!
    }
}

$list = $item->list ?? [];
var_dump($list); // array(0) {}  where is exception?
$list = $item->getList() ?? []; // exception - ok
```

### What is the expected result?

```php
$list = $item->list ?? []; // Exception division by zero
```

### What do you get instead?

```php
$list = $item->list ?? []; // array(0) {}
```

### Additional info

Reason in:
https://github.com/yiisoft/yii2/blob/master/framework/db/BaseActiveRecord.php#L336-L342

`??` - operator first make `isset()`, so it calls `__isset()`, which suppresses errors.

The problem is that it is very difficult to detect an error, because they are suppressed.

| Q                | A
| ---------------- | ---
| Yii version      | all
| PHP version      | 7.0
| Operating system | all

**Comment by samdark** [comment]
Any proposal on how to possibly fix that?

**Comment by paulzi** [comment]
@samdark how about this?

**Comment by SamMousa** [comment]
I agree with the fact that this is not desirable behavior, but changing it is a big break in backwards compatibility.

**Comment by paulzi** [comment]
I agree that changing behavior is dangerous, and can reveal some problems in existing projects. But complete suppression without logging is also bad.

**Comment by SamMousa** [comment]
I agree, logging is fine and doesn't change behavior. Or direct your PR at 3.0 and fix the behavior :-D
